modules/gapi/src/executor/gstreamingexecutor.cpp

```diff
@@ -553,6 +555,98 @@ void cv::gimpl::GStreamingExecutor::setSource(GRunArgs &&ins)
+        
+        auto xtract_in = [&](ade::NodeHandle slot_nh, std::vector<RcDesc> &vec) {
+            const auto orig_data_nh
+                = m_gim.metadata(slot_nh).get<DataSlot>().original_data_node;
+            const auto &orig_data_info
+                = gm.metadata(orig_data_nh).get<Data>();
+            if (orig_data_info.shape == GShape::GARRAY) {
+                // FIXME: GArray lost host constructor problem
+                GAPI_Assert(!cv::util::holds_alternative<cv::util::monostate>(orig_data_info.ctor));
+            }
+            vec.emplace_back(RcDesc{ orig_data_info.rc
+                                   , orig_data_info.shape
+                                   , orig_data_info.ctor});
+        };
+        auto xtract_out = [&](ade::NodeHandle slot_nh, std::vector<RcDesc> &vec, cv::GMetaArgs &metas) {
+            const auto orig_data_nh
+                = m_gim.metadata(slot_nh).get<DataSlot>().original_data_node;
+            const auto &orig_data_info
+                = gm.metadata(orig_data_nh).get<Data>();
+            if (orig_data_info.shape == GShape::GARRAY) {
+                // FIXME: GArray lost host constructor problem
+                GAPI_Assert(!cv::util::holds_alternative<cv::util::monostate>(orig_data_info.ctor));
+            }
+            vec.emplace_back(RcDesc{ orig_data_info.rc
+                                   , orig_data_info.shape
+                                   , orig_data_info.ctor});
+            metas.emplace_back(orig_data_info.meta);
+        };
+
+        for (auto& op : m_ops) {
+            op.isl_exec = m_gim.metadata(op.nh).get<IslandExec>().object;
+
+             std::vector<RcDesc> input_rcs;
+             std::vector<RcDesc> output_rcs;
+             cv::GMetaArgs output_metas;
+
+            input_rcs.reserve(op.nh->inNodes().size());
+            // in_constants.reserve(nh->inNodes().size()); // FIXME: Ugly
+            output_rcs.reserve(op.nh->outNodes().size());
+            output_metas.reserve(op.nh->outNodes().size());
+
+            for (auto in_slot_nh  : op.nh->inNodes())  xtract_in(in_slot_nh,  input_rcs);
+            for (auto out_slot_nh : op.nh->outNodes()) xtract_out(out_slot_nh, output_rcs, output_metas);
+
+            op.in_objects = input_rcs;
+            op.out_objects = output_rcs;
+            op.out_metas = output_metas;
+        }
+
+        wasFinished = true;
+    } else {
+        bool canReshape = true;
+        for (auto &op : m_ops) {
+            canReshape = op.isl_exec->canReshape();
+            if (!canReshape)
+                break;
+        }
+
+        if (canReshape) {
+            auto& g = *m_orig_graph.get();
+            ade::passes::PassContext ctx{g};
+            passes::initMeta(ctx, metas);
+            passes::inferMeta(ctx, true);
+            // outMetas()?
+            for (auto &op : m_ops)
+                op.isl_exec->reshape(g, comp_args);
+
+        } else {
+            //???
```

Comment:
Throw an error here.

Reply:
This code is removed.
Cases with extract meta data and reshape will be implemented in future versions